packages/devp2p/src/types.ts

```diff
@@ -0,0 +1,205 @@
+import type { DPT } from './dpt'
+import type { Common } from '@ethereumjs/common'
+import type { Socket } from 'net'
+
+interface ProtocolConstructor {
+  new (...args: any[]): any
+}
+
+export interface Capabilities {
+  name: string
+  version: number
+  length: number
+  constructor: ProtocolConstructor
+}
+
+export enum DISCONNECT_REASON {
+  DISCONNECT_REQUESTED = 0x00,
+  NETWORK_ERROR = 0x01,
+  PROTOCOL_ERROR = 0x02,
+  USELESS_PEER = 0x03,
+  TOO_MANY_PEERS = 0x04,
+  ALREADY_CONNECTED = 0x05,
+  INCOMPATIBLE_VERSION = 0x06,
+  INVALID_IDENTITY = 0x07,
+  CLIENT_QUITTING = 0x08,
+  UNEXPECTED_IDENTITY = 0x09,
+  SAME_IDENTITY = 0x0a,
+  TIMEOUT = 0x0b,
+  SUBPROTOCOL_ERROR = 0x10,
+}
+export type DNSOptions = {
+  /**
+   * ipv4 or ipv6 address of server to pass to native dns.setServers()
+   * Sets the IP address of servers to be used when performing
+   * DNS resolution.
+   * @type {string}
+   */
+  dnsServerAddress?: string
+}
+
+export interface DPTOptions {
+  /**
+   * Timeout for peer requests
+   *
+   * Default: 10s
+   */
+  timeout?: number
+
+  /**
+   * Network info to send a long a request
+   *
+   * Default: 0.0.0.0, no UDP or TCP port provided
+   */
+  endpoint?: PeerInfo
+
+  /**
+   * Function for socket creation
+   *
+   * Default: dgram-created socket
+   */
+  createSocket?: Function
+
+  /**
+   * Interval for peer table refresh
+   *
+   * Default: 60s
+   */
+  refreshInterval?: number
+
+  /**
+   * Toggles whether or not peers should be queried with 'findNeighbours'
+   * to discover more peers
+   *
+   * Default: true
+   */
+  shouldFindNeighbours?: boolean
+
+  /**
+   * Toggles whether or not peers should be discovered by querying EIP-1459 DNS lists
+   *
+   * Default: false
+   */
+  shouldGetDnsPeers?: boolean
+
+  /**
+   * Max number of candidate peers to retrieve from DNS records when
+   * attempting to discover new nodes
+   *
+   * Default: 25
+   */
+  dnsRefreshQuantity?: number
+
+  /**
+   * EIP-1459 ENR tree urls to query for peer discovery
+   *
+   * Default: (network dependent)
+   */
+  dnsNetworks?: string[]
+
+  /**
+   * DNS server to query DNS TXT records from for peer discovery
+   */
+  dnsAddr?: string
+}
+
+export interface DPTServerOptions {
+  /**
+   * Timeout for peer requests
+   *
+   * Default: 10s
+   */
+  timeout?: number
+
+  /**
+   * Network info to send a long a request
+   *
+   * Default: 0.0.0.0, no UDP or TCP port provided
+   */
+  endpoint?: PeerInfo
+
+  /**
+   * Function for socket creation
+   *
+   * Default: dgram-created socket
+   */
+  createSocket?: Function
+}
+
+export enum EthProtocol { // What does this represent?
```

Comment:
Agree with this comment slightly. The `enum` name is weird given one of the options is `eth`. Would it be bad to rename it to just `Protocol` like we do in `client`?

Reply:
Renaming to protocol would cause conflicts as we already have a Protocol class, but I've renamed it to ProtocolLabel, which I feel is clearer.

Reply:
We actually also have another interface called Capabilities!
```
export interface Capabilities {
  name: string
  version: number
  length: number
  constructor: ProtocolConstructor
}

``